examples/classifcation/text_classification.exs

```diff
@@ -0,0 +1,273 @@
+# Based on https://valueml.com/named-entity-recognition-using-lstm-in-keras/
+
+# References
+# http://alexminnaar.com/2019/08/22/ner-rnns-tensorflow.html
+
+# Test Data: https://github.com/bhuvanakundumani/NER_tensorflow2.2.0/blob/master/data/test.txt
+
+# Train Data: https://github.com/bhuvanakundumani/NER_tensorflow2.2.0/blob/master/data/train.txt
+
+# Validation Data: https://github.com/bhuvanakundumani/NER_tensorflow2.2.0/blob/master/data/valid.txt
+
+Mix.install([
+  {:axon, github: "elixir-nx/axon"},
+  {:nx, "~> 0.2.1"},
+  {:exla, "~> 0.2.2"}
+])
+
+EXLA.set_as_nx_default([:tpu, :cuda, :rocm, :host])
+
+defmodule TextClassification do
+  require Axon
+  # From Example http://alexminnaar.com/2019/08/22/ner-rnns-tensorflow.html
+  EXLA.set_as_nx_default([:tpu, :cuda, :rocl, :host]) # My :host option doesnt work, jit_apply is undefined
+  require Logger
+  @sequence_length 75
+  @batch_size 128
+  @lstm 128
+  @embed_dimension 256
+  @def_split " "
+  @start "-DOCSTART- -X- -X- O\n"
+  @ending "-------------------------------------------------------------"
+  @incorrect [".", "?", "%", ",", ":", ";", "-docstart-"]
+
+  # seqlen
+  # word_count
+
+  def train(model, data) do
+    model
+    |> Axon.Loop.trainer(&kl_divergence/2, Axon.Optimizers.adam(0.001))
+    |> Axon.Loop.metric(:accuracy, "Accuracy")
+    |> Axon.Loop.run(data, %{}, epochs: 3, iterations: 5)
+
+  end
+
+  defp kl_divergence(y_true, y_pred) do
+    Axon.Losses.kl_divergence(y_true, y_pred, reduction: :mean)
+  end
+
+  def build_model(word_count, label_count) do
+      Axon.input({nil, @sequence_length, 1}, "inputs")
+       |> Axon.embedding(word_count, @sequence_length)
+       |> Axon.nx(fn t ->
+         t[[0..-1//1, -1]] # Dropping out the sized 1 column created from the embed
+       end)
+       |> Axon.spatial_dropout(rate: 0.1)
+       |> Axon.lstm(@lstm)
+       |> then(fn {{new_cell, new_hidden}, out} ->
+         out
+       end)
+      |> Axon.dropout(rate: 0.2)
+      |> Axon.dense(label_count, activation: :softmax)
+  end
+
+  defp encode(dictionary, word) do
+    with {:ok, encoded} <- Map.fetch(dictionary, word) do
+      encoded
+    else
+      _->
+      0
+    end
+  end
+
+  defp decode(dictionary, word) do
+    with {:ok, decoded} <- Map.fetch(dictionary, word) do
+      decoded
+    else
+      _-> "Unknown"
+    end
+  end
+
+  def transform_words(word_labels, word_to_idx, label_to_idx, wcount, lcount) do
+
+    train_data =
+      word_labels
+      |> Enum.map(fn {word, label} ->
+        encode(word_to_idx, word)
+      end)
+      |> Enum.chunk_every(@sequence_length, 1, :discard)
+      |> Enum.drop(-1)
+      |> Nx.tensor
+      |> Nx.divide(wcount)
+      |> Nx.reshape({:auto, @sequence_length, 1})
```

Comment:
Relates to shape comment above :)

```suggestion
      |> Nx.reshape({:auto, @sequence_length})
```

Reply:
So this was a weird bit for me, I followed alot of tutorials at the start for the basic concept, and the sizing were always not well explained. I'll do as suggested above, as it makes more sense, I believe the problem at the start I was having was due to the embedding layer and me not understanding I needed to squeeze it out earlier.